pkg/internal/export/otel/metrics.go

```diff
@@ -169,6 +166,16 @@ func newMetricsReporter(ctx context.Context, cfg *MetricsConfig, ctxInfo *global
 	return &mr, nil
 }
 
+func (r *Metrics) release() error {
```

Comment:
Hmmmm, I think I need to make a test for this to be 100% certain. Essentially, we have the following issue:

1. The tracers are per `svc.ID` and we make them on first use and keep them in the LRU cache in `ReporterPool`.
2. If we have more than 16 services reporting data, we don't have enough entries in the pool and reporters must be evicted. On eviction we shutdown the reporter that's evicted, but there's no guarantee we have sent everything we had there.
3. We see this in the logs:
```
time=2023-12-20T21:04:32.486Z level=DEBUG msg="evicting metrics reporter from cache" component=otel.TracesReporter service="{Name:livenessprobe Namespace: SDKLanguage:go Instance:beyla-n4h9q-3925388}"
2023/12/20 21:04:32 failed to upload metrics: gRPC exporter is shutdown
```
That last message comes from the OTel SDK, when metrics are committed.

So to solve this, I essentially changed the code to not shutdown the reported on eviction, but when the reporter is to be disposed by the GC. Once we evict it from the LRU, it will only be kept live by the batching logic in the exporter and once all events are reported I think the finalizer will shutdown the connection.

Reply:
Oh yeah! I misunderstood the cause of the initial problem. In that case, I think the current PR is fine. Even, does it have sense to use a runtime finalizer?

Reply:
Hey @mariomac, I modified our system wide instrumentation test to reduce the LRU cache to 1, which forces cache eviction immediately in our test. So my fix to use finalizers didn't work. Essentially, when any MetricProvider is shutdown, it also shutdown the underlying exporter! I don't know why the OTel SDK is designed like this, but definitely happens. So I changed the code to forceflush on provider LRU expulsion, since they all seem to share the exporter.

Expanding the size of the LRU is generally good for performance, but we can't shutdown the providers it seems. I'm not sure what it all does that regular GC cleanup won't do. 

I'm not sure how do we verify if all resources held by a metric provider are cleaned up on GC of a provider?

Reply:
I confirmed, calling Shutdown() on the provider calls `func (r *PeriodicReader) Shutdown(ctx context.Context) error {` which calls `sErr := r.exporter.Shutdown(ctx)`. It does nothing else that specific to the metric provider as far as I can see.

Reply:
I think the current solution is perfect!! Let's rely on the GC to cleanup all the stuff.